src/shared-components/alert/index.tsx

```diff
@@ -61,45 +57,37 @@ type AlertState = {
  *
  * All alerts are dimissable by clicking on them. However, you can use the `duration` prop to determine if the alert is sticky or dismissed on a timer (in units of seconds).
  */
-export class Alert extends React.Component<AlertProps, AlertState> {
-  static propTypes = {
-    avatarSrc: PropTypes.string,
-    content: PropTypes.node.isRequired,
-    ctaContent: PropTypes.node,
-    duration: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    truncateText: PropTypes.bool,
-    type: PropTypes.oneOf(['success', 'error', 'default']),
-    onExit: PropTypes.func,
-  };
-
-  static defaultProps = {
-    avatarSrc: '',
-    ctaContent: null,
-    duration: 3,
-    truncateText: false,
-    type: 'default',
-    onExit: () => undefined,
-  };
-
-  static Container = ({ children }: { children: React.ReactNode }) => (
-    <AlertsContainer>{children}</AlertsContainer>
-  );
-
-  contentText = React.createRef<HTMLDivElement>();
+export const Alert = ({
+  avatarSrc = '',
+  content,
+  ctaContent = null,
+  duration = 3,
+  onExit = () => undefined,
+  truncateText = false,
+  type = 'default',
+  ...rest
+}: AlertProps) => {
+  const [exiting, setExiting] = useState(false);
+  const [exited, setExited] = useState(false);
+
+  const contentText = React.createRef<HTMLDivElement>();
+
+  let timer: number | undefined;
+
+  const alertExitHandler = () => {
+    setExiting(true);
+    window.clearTimeout(timer);
 
-  timer: number | undefined;
-
-  state = {
-    exiting: false,
-    exited: false,
+    window.setTimeout(() => {
+      setExited(true);
+      onExit();
```

Comment:
The old diff has it like so: `onExit({ ...rest });`.

Reply:
Yeah that was on purpose, This logic didnt make sense  to me and is not used anywhere else in Radiance. Then going to Curology I couldnt find any reference that they are using the onExit params. I have a commit ready to revert back this change and leave it as it was, what do you think we should remove the onExit params or restore?

Reply:
I just restored the old signature.

Reply:
@daigof I'm okay if API changes are made after investigating and seeing that we're not using it, but in general I'd prefer more functional changes (like this one, compared to removing an optional onClick that we provide that does nothing and instead making it require) be done in separate PRs so the change is not lost in the noise, and so we can track it more clearly in the `CHANGELOG`. You can merge in this PR either way, just want to make sure we're documenting changes like this.